### PR TITLE
Split `ContentViewFilter` in to several classes

### LINKED ISSUE #92: Content view filter creation errored
Issue:
```python
>>> cvf = ContentViewFilter(content_view=cv,type='erratum',name='filter1').create(True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/../virt_env/virt1/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 719, in create
    return self.read(attrs=self.create_json(create_missing))
  File "/../virt_env/virt1/lib/python2.7/site-packages/nailgun/entities.py", line 924, in read
    return super(ContentViewFilter, self).read(entity, attrs, ignore)
  File "/../virt_env/virt1/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 562, in read
    setattr(entity, field_name, attrs[field_name])
KeyError: 'original_packages'
```

Possible required fixes:
* original_packages is required only for package type filters. So may be 
'original_packages': entity_fields.BooleanField(null=True)?
* add logic to handle the fact that, this field is required only for package type filters.
* update the 'type' field to handle the choices (rpm, package_group, erratum)
* we need the logic to handle each 'type' assuming that different fields are used for different types

### PR DESCRIPTION
**Pull request by Ichimonji10**
Fix #92: "Content view filter creation errored".

Example of new classes in use:

```python
>>> from nailgun import entities
>>> [entities.AbstractContentViewFilter(id=id).read().type for id in range(8, 11)]
['erratum', 'package_group', 'rpm']
>>> entities.ErratumContentViewFilter(id=8).read().type
'erratum'
>>> entities.PackageGroupContentViewFilter(id=9).read().type
'package_group'
>>> entities.RPMContentViewFilter(id=10).read().type
'rpm'
>>> rpm_cv_filter = entities.RPMContentViewFilter().create(True)
>>> rpm_cv_filter.type, rpm_cv_filter.original_packages
('rpm', False)
```

### CODE REVIEW

File: nailgun/entities.py [hunk 1]
```diff
@@ -924,34 +924,74 @@ def read(self, entity=None, attrs=None, ignore=('content_view_filter',)):
...
+            **kwargs
+        )
+        self._fields['type'].default = 'package_group'
+
+
+class RPMContentViewFilter(AbstractContentViewFilter):
+    """A representation of a Content View Filter of type "rpm"."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {'original_packages': entity_fields.BooleanField()}
```
Comment (omaciel) [inline nailgun/entities.py]: Why is this needed?

**Comment by omaciel** [comment]
ACK pending clarification on code change

**Comment by Ichimonji10** [comment]
```python
>>> [entities.AbstractContentViewFilter(id=id).read().type for id in range(4, 7)]
['erratum', 'rpm', 'package_group']
>>> for id in range(4, 7):
...     entities.AbstractContentViewFilter(id=id).read_json().keys()
...
dict_keys(['type', 'id', 'content_view', 'description', 'repositories', 'created_at', 'rules', 'updated_at', 'name', 'inclusion']),
dict_keys(['type', 'id', 'content_view', 'description', 'repositories', 'created_at', 'rules', 'updated_at', 'name', 'inclusion', 'original_packages']),
dict_keys(['type', 'id', 'content_view', 'description', 'repositories', 'created_at', 'rules', 'updated_at', 'name', 'inclusion'])
```

File: nailgun/entities.py (hunk 1 above)
Comment (Ichimonji10) [inline nailgun/entities.py]:
```python
>>> [entities.AbstractContentViewFilter(id=id).read().type for id in range(4, 7)]
['erratum', 'rpm', 'package_group']
>>> for id in range(4, 7):
...     entities.AbstractContentViewFilter(id=id).read_json().keys()
...
dict_keys(['type', 'id', 'content_view', 'description', 'repositories', 'created_at', 'rules', 'updated_at', 'name', 'inclusion']),
dict_keys(['type', 'id', 'content_view', 'description', 'repositories', 'created_at', 'rules', 'updated_at', 'name', 'inclusion', 'original_packages']),
dict_keys(['type', 'id', 'content_view', 'description', 'repositories', 'created_at', 'rules', 'updated_at', 'name', 'inclusion'])
```

Comment (omaciel) [inline nailgun/entities.py]: Cool. I should have been more explicit :) I wanted to see a comment added to the code so that anyone can see this :)

Comment (Ichimonji10) [inline nailgun/entities.py]: Ahh, I understand! I'll do that.

**Comment by omaciel** [comment]
ACK

**Comment by sthirugn** [comment]
looks great ACK
